docs/source/tutorial/basic.rst

```diff
@@ -2,21 +2,586 @@
+
+.. accordion:: Install localhost computer and code
+
+    Let's begin by setting up the computer using the ``verdi computer`` subcommand:
+
+    .. code-block:: console
+
+        $ verdi computer setup -L tutor -H localhost -T local -S direct -w `echo $PWD/work` -n
+        $ verdi computer configure local tutor --safe-interval 5 -n
+
+    The first commands sets up the computer with the following options:
+
+    * *label* (``-L``): tutor
+    * *hostname* (``-H``): localhost
+    * *transport* (``-T``): local
+    * *scheduler* (``-S``): direct
+    * *work-dir* (``-w``): The ``work`` subdirectory of the current directory
+
+    The second command *configures* the computer with a minimum interval between connections (``--safe-interval``) of 5 seconds.
+    For both commands, the *non-interactive* option (``-n``) is added to not prompt for extra input.
+
+    Next, let's set up the code we're going to use for the tutorial:
+
+    .. code-block:: console
+
+        $ verdi code setup -L add --on-computer --computer=tutor -P arithmetic.add --remote-abs-path=/bin/bash -n
+
+    This command sets up a code with *label* ``add`` on the *computer* ``tutor``, using the *plugin* ``arithmetic.add``.
+
+.. note::
+    A typical real-world example of a computer is a remote supercomputing facility.
+    Codes can be anything from a Python script to powerful *ab initio* codes such as Quantum Espresso or machine learning tools like Tensorflow.
+
+Let's have a look at the codes that are available to us:
+
+.. code:: console
+
+    $ verdi code list
+
+You can see a single code ``add@tutor``, with PK = 5, in the printed list.
+This code allows us to add two integers together.
+The ``add@tutor`` identifier indicates that the code with label ``add`` is run on the computer with label ``tutor``.
+To see more details about the computer, you can use the following ``verdi`` command:
+
+.. code:: console
+
+    $ verdi computer show tutor
+
+Note that the *Work directory* has been set up as the ``work`` subdirectory of the current directory.
+This is the directory in which the calculations running on the ``tutor`` computer will be executed.
+
+Let's now start up the ``verdi shell`` again and load the ``add@tutor`` code using its label:
+
+.. code-block:: ipython
 
-See `issue #3981 <https://github.com/aiidateam/aiida-core/issues/3981>`_.
+    In [1]: code = load_code(label='add')
 
+Every code has a convenient tool for setting up the required input, called the builder.
+It can be obtained by using the ``get_builder`` method:
+
+.. code-block:: ipython
+
+    In [2]: builder = code.get_builder()
+
+Using the builder, you can easily set up the calculation by directly providing the input arguments.
+Let's use the ``Int`` node that was created by our previous ``calcfunction`` as one of the inputs:
+
+.. code-block:: ipython
+
+    In [3]: builder.x = load_node(pk=4)
```

Comment:
Is there a good way to figure out which PK to use here in case that mine differs from the tutorial?

Reply:
Perhaps a good way to help the user load the output node from the calculation function in case they have forgotten the PK is by having them copy the UUID from the provenance chart they have generated and use e.g.

```
In [3]: builder.x = load_node(uuid='42541d38')
```

That way they can immediately see that you can also use the UUID to load a node. I suppose it might be confusing that you don't use the _whole_ UUID, but we can explain that as well.

Maybe there is a much easier way that I'm missing though.

Reply:
No, I think it's worth expounding on the relationship between PKs and UUIDs just a little bit. So I am in favor of this change.